xmppserver/src/main/java/org/jivesoftware/openfire/OfflineMessageStore.java

```diff
@@ -563,4 +619,154 @@ static boolean shouldStoreMessage(final Message message) {
         }
         return true;
     }
+
+    private void setTimer() {
+        cancelTimer();
+
+        timer = new Timer(true);
+        timer.schedule(new TimerTask() {
+            @Override
+            public void run() {
+                try {
+                    if (ClusterManager.isClusteringStarted())
+                    {
+                        if (ClusterManager.isSeniorClusterMember())
+                        {
+                            if (deleteOldOfflineMessagesFromDB())
+                            {
+                                readSizeForAllUsers();
+                            }
+                        }
+                    }
+                    else
+                    {
+                        if (deleteOldOfflineMessagesFromDB())
+                        {
+                            readSizeForAllUsers();
+                        }
+                    }
+                } catch (Exception e) {
+                    Log.error("Offline message cleaning - Could not set timer for check interval!", e.fillInStackTrace());
```

Comment:
Why do you use `Log.error("my message", e.fillInStackTrace());` here? Is that different from just doing `Log.error("my message", e);` ?